File: easy/1480-running-sum-of-1d-array/running-sum-of-1d-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    let sums = [];
    let runningSum = nums[0];
    sums[0] = runningSum;
    for(let i = 1; i<nums.length; i++){
        runningSum += nums[i];
        sums.push(runningSum);
    }
    return sums
};

// solution using map
var runningSum = function(nums) {
    let runningSum = nums[0];
    return nums.map((num) => runningSum += num);
};

File: easy/1480-running-sum-of-1d-array/running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {
        int[] sums = new int[nums.length];
        // int[] sums = new int[]; // Java does not allow dynamic array sizing, so this would not work
        int runningSum = 0;
        sums[0] = runningSum;
        for (int i=0; i<nums.length; i++){
            runningSum += nums[i];
            sums[i] = runningSum;
        }
        return sums;
    }
}